src/features/CreateEntities/ui/movies/MoviesForm.tsx

```diff
@@ -0,0 +1,288 @@
+import { FC, useEffect } from 'react';
+
+import { InputLabel } from '@mui/material';
+import { useQueryClient } from '@tanstack/react-query';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+
+import { useMovieMutation } from 'entities/movie/hooks/useCreateOrUpdateMovieMutation';
+import { MOVIE_GENRES } from 'shared/constants/movieGenres';
+import { MAX_IMAGE_SIZE } from 'shared/constants/utils';
+import { useTranslation } from 'shared/hooks/i18nHook';
+import {
+    Alert,
+    Button,
+    Checkbox,
+    Dialog,
+    LoadingSpinner,
+    MenuItem,
+    Select,
+    TextField
+} from 'shared/ui';
+
+import { FormFields, SelectedMovie } from './types';
+
+interface MoviesFormProps {
+    open: boolean;
+    onClose: () => void;
+    afterClose: () => void;
+    selectedMovie: SelectedMovie | null;
+}
+
+const defaultValues: FormFields = {
+    title: '',
+    description: '',
+    duration: '',
+    image: null,
+    genre: '',
+    releaseDate: '',
+    trended: false
+};
+
+export const MoviesForm: FC<MoviesFormProps> = ({ open, onClose, selectedMovie, afterClose }) => {
+    const { mutate: createOrUpdateMovieMutation, isPending, isError, error } = useMovieMutation();
+    const queryClient = useQueryClient();
+    const { t } = useTranslation('common');
+
+    const { handleSubmit, control, reset, setValue } = useForm({
+        defaultValues,
+        mode: 'onBlur'
+    });
+
+    const onSubmit: SubmitHandler<FormFields> = async ({
+        title,
+        description,
+        genre,
+        image,
+        releaseDate,
+        duration,
+        trended
+    }) => {
+        createOrUpdateMovieMutation(
```

Comment:
on this level you'll decided do we need to create or update

Reply:
in such case you will have 2 separate business actions, and in future in case when you add super_admin for example, you can provide him ability to create/update movies, but for common admins only create. It's just an example, but I try to explain the idea behind separation